code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -106,6 +106,13 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 	var/cached_gases = gases
 	TOTAL_MOLES(cached_gases, .)
 
+	/// Checks to see if gas amount exists in mixture
+	/// WARNING - DO NOT USE THIS IN PERFORMANCE INTENSIVE CODE!!!
+/datum/gas_mixture/proc/has_gas(gas_id, amount=0)
+	ASSERT_GAS(gas_id, src)
+	var/list/cached_gases = gases
+	return amount < cached_gases[gas_id][MOLES]
```

Comment:
For one access caching isn't worth it. Store the result of the check as a var, and call garbage_collect() before returning so gaslists don't get a lot of empty entries.

Reply:
942c867 does all the above.  Also good to know that `garbage_collect` needs to be called whenever `ASSERT_GAS` is used.